src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,197 @@
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext webApplicationContext
+    ) throws SQLException {
+
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/book/create-table-books_categories.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/category/add-two-categories.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/book/add-two-books.sql")
+            );
+        }
+    }
+
+    @AfterAll
+    static void afterAll(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/book/remove-all-books.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/category/remove-all-categories.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Save new book.")
+    void createBook_ValidRequestDto_Ok() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setTitle("Effective Java");
+        requestDto.setAuthor("Joshua Bloch");
+        requestDto.setIsbn("978-0134685991");
+        requestDto.setPrice(BigDecimal.valueOf(49.99));
+        requestDto.setDescription("A comprehensive guide to best practices in Java programming.");
+        requestDto.setCategoryIds(List.of(1L, 2L));
+
+        BookDto expected = new BookDto();
+        expected.setTitle(requestDto.getTitle());
+        expected.setAuthor(requestDto.getAuthor());
+        expected.setIsbn(requestDto.getIsbn());
+        expected.setPrice(requestDto.getPrice());
+        expected.setDescription(requestDto.getDescription());
```

Comment:
better move such logic to the private method or util class, and simplify your test cases